aplib.net-demo/Assets/Testing/AplibTests/SolveMazeTest.cs

```diff
@@ -0,0 +1,196 @@
+using System.Collections;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+using Aplib.Core;
+using Aplib.Core.Belief;
+using Aplib.Core.Desire;
+using Aplib.Core.Desire.Goals;
+using Aplib.Core.Intent.Actions;
+using Aplib.Core.Intent.Tactics;
+using Aplib.Integrations.Unity;
+using UnityEngine.SceneManagement;
+
+public class SolveMazeBeliefSet : BeliefSet
+{
+    /// <summary>
+    /// The player object in the scene.
+    /// </summary>
+    public Belief<GameObject, GameObject> Player = new(GameObject.Find("Player"), x => x);
+
+    /// <summary>
+    /// Whether or not an obstacle is in front of the player, preventing it to move forward.
+    /// </summary>
+    public Belief<Transform, bool> IsObstacleInFront = new
+        (
+            reference: GameObject.Find("Player").transform,
+            getObservationFromReference: playerTransform =>
+            {
+                // Determine the eye position of the player, since the player transform is at the feet of the player.
+                // Also determine the direction that the player should look in.
+                Vector3 playerPosition = playerTransform.position;
+                Vector3 eyePosition = playerPosition + new Vector3(0, 1.75f, 0);
+                Vector3 lookFrontDirection = playerPosition - eyePosition + playerTransform.forward;
+                Vector3 lookRightDirection = playerPosition - eyePosition + (playerTransform.forward + 0.5f * playerTransform.right);
+
+                // Shoot two raycasts from the eye of the player. We shoot two so that the player has some margin to move.
+                bool isHitInFront = Physics.Raycast(eyePosition, lookFrontDirection, out RaycastHit hitFront, 4f);
+                bool isHitToRight = Physics.Raycast(eyePosition, lookRightDirection, out RaycastHit hitRight, 4f);
+
+                Debug.DrawRay(eyePosition, lookFrontDirection, Color.red);
+                Debug.DrawRay(eyePosition, lookRightDirection, Color.red);
+
+                // Check if there is a hole in front of the player.
+                bool isHoleInFront = !isHitInFront && !isHitToRight;
+
```

Comment:
Why is this called a hole, and not just 'nothing'? And why can't there be anything on the right in order to have a hole in the front?